### PR TITLE
TEC-3285, EBT-77: Extend TEC views for Aggregator client #305

### PR DESCRIPTION
**Pull request by mitogh**
Remove EBT logic from the aggregator templates to leverage that logic into EBT plugin instead.

Add new filter to simplify the logic to load new files on aggregator admin views.

Tickets: 

- [TEC-3285]
- [EBT-77]

[TEC-3285]: https://moderntribe.atlassian.net/browse/TEC-3285
[EBT-77]: https://moderntribe.atlassian.net/browse/EBT-77

### CODE REVIEW

File: src/Tribe/Aggregator/Page.php [hunk 1]
```diff
@@ -330,6 +329,18 @@ public function template( $name, $data = array(), $echo = true ) {
...
+		 *
+		 * @since TBD
+		 *
+		 * @param string       $file Complete path to include the PHP File.
+		 * @param string|array $name Template name.
+		 * @param array        $data The Data that will be used on this template.
+		 *
+		 * @return string Complete path to the file to be included.
+		 */
+		$file = apply_filters( "tribe_aggregator_template_file/{$file_name}", $file, $name, $data );
```
Comment (tr1b0t) [inline src/Tribe/Aggregator/Page.php]: :warning: Words in hook names should be separated using underscores. Expected: "tribe_aggregator_template_file_{$file_name}", but found: "tribe_aggregator_template_file/{$file_name}".

<sub>hash: 07f84e9f5532c72ed26e8de6bb80f1ad</sub>

File: src/admin-views/aggregator/origins/refine.php [hunk 2]
```diff
@@ -57,24 +44,25 @@
  *
  * @param array $keyword_exclusions List of origins excluded.
  */
-$keyword_exclusions = json_encode( apply_filters( 'tribe_events_aggregator_refine_keyword_exclusions', $keyword_exclusions ) );
+$keyword_exclusions = apply_filters( 'tribe_events_aggregator_refine_keyword_exclusions', [ 'facebook' ] );
+$keyword_exclusions = json_encode( $keyword_exclusions );
```
Comment (tr1b0t) [inline src/admin-views/aggregator/origins/refine.php]: :warning: json_encode() is discouraged. Use wp_json_encode() instead.

<sub>hash: acf70da3daa51e852c0f36dc01930918</sub>

File: src/admin-views/aggregator/origins/refine.php [hunk 3]
```diff
@@ -57,24 +44,25 @@
...
  *
  * @since 4.6.24
  *
  * @param array $location_exclusions List of origins excluded.
  */
-$location_exclusions = json_encode( apply_filters( 'tribe_events_aggregator_refine_location_exclusions', $location_exclusions ) );
-
-
+$location_exclusions = apply_filters( 'tribe_events_aggregator_refine_location_exclusions', [ 'url', 'facebook' ] );
+$location_exclusions = json_encode( $location_exclusions );
```
Comment (tr1b0t) [inline src/admin-views/aggregator/origins/refine.php]: :warning: json_encode() is discouraged. Use wp_json_encode() instead.

<sub>hash: 97184ad1e6b46cf48bf27abb91f0f458</sub>

File: src/admin-views/aggregator/origins/refine.php [hunk 4]
```diff
@@ -85,12 +73,13 @@ class="tribe-ea-field tribe-ea-size-xlarge"
...
+		<?php if ( 'eventbrite' !== $origin_slug ) : ?>
 		<div class="tribe-refine">
 			<?php
-				$start = empty( $record->meta['start'] ) ? '' : $record->meta['start'];
-				if ( is_numeric( $start ) ) {
-					$start = date( Tribe__Date_Utils::DATEONLYFORMAT, $start );
-				}
+			$start = empty( $record->meta['start'] ) ? '' : $record->meta['start'];
+			if ( is_numeric( $start ) ) {
+				$start = date( Tribe__Date_Utils::DATEONLYFORMAT, $start );
```
Comment (tr1b0t) [inline src/admin-views/aggregator/origins/refine.php]: :no_entry_sign: date() is affected by runtime timezone changes which can cause date/time to be incorrectly displayed. Use gmdate() instead.

<sub>hash: 2f308f0bf1c220a90f965452622519d5</sub>

File: src/admin-views/aggregator/origins/refine.php [hunk 5]
```diff
@@ -141,4 +128,4 @@ class="tribe-ea-field tribe-ea-dropdown tribe-ea-size-medium"
 			</select>
 		</div>
 	</td>
-</tr>
+</tr>
```
Comment (tr1b0t) [inline src/admin-views/aggregator/origins/refine.php]: :no_entry_sign: File must end with a newline character

<sub>hash: db6d368f7afe773b13e2c9256c74eb1c</sub>

File: readme.txt [hunk 6]
```diff
@@ -224,12 +224,14 @@ Remember to always make a backup of your database and files before updating!
...
 * Tweak - Minify the Freemius svg assets. [TEC-3215]
 * Tweak - Remove "(beta)" label from URL source type of import. [TEC-3289]
+* Tweak - Remove Eventbrite dependency on Aggregator Admin view. [EA-133]
 * Fix - Breakpoint code compatibility with the latest version of jQuery 3.4.1 [BTRIA-154]
 * Fix - Remove numbers from listed events in event list widget for Genesis theme. [TEC-2957]
 * Fix - Customizer accent color affects featured line and text. [TEC-3277]
 * Fix - Resolve conflict between datepicker holding consistent size when navigating and flexibility on small screen sizes. [TEC-3322]
 * Fix - Respect EA `post_status` from settings when an event does not have a defined value. [TEC-3255]
 * Fix - Save the `post_status` for a scheduled import when using a different value other than `publish` [TEC:3296]
+* Hooks - More refine filter that is used to locate the aggregator admin files `tribe_aggregator_template_file/{$file_name}`. [EA-133]
```
Comment (lucatume) [inline readme.txt]: This entry is not clear, could you please rephrase it?

File: src/Tribe/Aggregator/Page.php [hunk 7]
```diff
@@ -317,10 +316,10 @@ public function register_menu_item() {
 	 */
 	public function template( $name, $data = array(), $echo = true ) {
 		// Clean this Variable
-		$name = array_map( 'sanitize_title_with_dashes', (array) explode( '/', $name ) );
+		$name = array_map( 'sanitize_title_with_dashes', (array) explode( DIRECTORY_SEPARATOR, $name ) );
```
[the remark this answers is not in the document]
Comment (mitogh) [reply src/Tribe/Aggregator/Page.php]: This is mainly to make it consistent with: 

- https://github.com/moderntribe/the-events-calendar/blob/task/eventbrite-aggregator-view/src/Tribe/Aggregator/Page.php#L321

As is the value used to join the same lines

[the remark this answers is not in the document]
Comment (bordoni) [reply src/Tribe/Aggregator/Page.php]: We use `DIRECTORY_SEPARATOR` in a couple of places, we should stick to it.

**Comment by mitogh** [comment]
Closed the branch due `common` was merged into this PR.

PR opened instead: https://github.com/moderntribe/the-events-calendar/pull/3101

**Pull request closed without merging**